### ISSUE TITLE
Warnings due to new OGR field types in GDAL 2.0dev

### ISSUE THREAD

**Issue by rouault** [mapnik/mapnik#2685]
Compiling against latest GDAL 2.0dev, after http://trac.osgeo.org/gdal/wiki/rfc31_ogr_64 , gives :

```
g++ -o plugins/input/ogr/ogr_datasource.os -c -std=c++11 -Wall -pthread -ftemplate-depth-300 -Wno-pragmas -O3 -fPIC -DSHAPE_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DBOOST_REGEX_HAS_ICU -DHAVE_JPEG -DMAPNIK_USE_PROJ4 -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -Ideps -Ideps/clipper/include -Ideps/agg/include -Iinclude -I/home/even/mapbox/prefix/include -I/usr/include -I/usr/include/freetype2 -I/usr/include/libxml2 plugins/input/ogr/ogr_datasource.cpp
plugins/input/ogr/ogr_datasource.cpp: In member function ‘void ogr_datasource::init(const mapnik::parameters&)’:
plugins/input/ogr/ogr_datasource.cpp:333:20: warning: enumeration value ‘OFTInteger64’ not handled in switch [-Wswitch]
             switch (type_oid)
                    ^
plugins/input/ogr/ogr_datasource.cpp:333:20: warning: enumeration value ‘OFTInteger64List’ not handled in switch [-Wswitch]
plugins/input/ogr/ogr_datasource.cpp:333:20: warning: enumeration value ‘OFTMaxType’ not handled in switch [-Wswitch]
g++ -o plugins/input/ogr/ogr_featureset.os -c -std=c++11 -Wall -pthread -ftemplate-depth-300 -Wno-pragmas -O3 -fPIC -DSHAPE_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DBOOST_REGEX_HAS_ICU -DHAVE_JPEG -DMAPNIK_USE_PROJ4 -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -Ideps -Ideps/clipper/include -Ideps/agg/include -Iinclude -I/home/even/mapbox/prefix/include -I/usr/include -I/usr/include/freetype2 -I/usr/include/libxml2 plugins/input/ogr/ogr_featureset.cpp
g++ -o plugins/input/ogr/ogr_index_featureset.os -c -std=c++11 -Wall -pthread -ftemplate-depth-300 -Wno-pragmas -O3 -fPIC -DSHAPE_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DBOOST_REGEX_HAS_ICU -DHAVE_JPEG -DMAPNIK_USE_PROJ4 -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -Ideps -Ideps/clipper/include -Ideps/agg/include -Iinclude -I/home/even/mapbox/prefix/include -I/usr/include -I/usr/include/freetype2 -I/usr/include/libxml2 plugins/input/ogr/ogr_index_featureset.cpp
plugins/input/ogr/ogr_index_featureset.cpp: In instantiation of ‘mapnik::feature_ptr ogr_index_featureset<filterT>::next() [with filterT = mapnik::filter_in_box; mapnik::feature_ptr = std::shared_ptr<mapnik::feature_impl>]’:
plugins/input/ogr/ogr_index_featureset.cpp:193:16:   required from here
plugins/input/ogr/ogr_index_featureset.cpp:140:13: warning: enumeration value ‘OFTInteger64’ not handled in switch [-Wswitch]
             switch (type_oid)
             ^
plugins/input/ogr/ogr_index_featureset.cpp:140:13: warning: enumeration value ‘OFTInteger64List’ not handled in switch [-Wswitch]
plugins/input/ogr/ogr_index_featureset.cpp:140:13: warning: enumeration value ‘OFTMaxType’ not handled in switch [-Wswitch]
plugins/input/ogr/ogr_index_featureset.cpp: In instantiation of ‘mapnik::feature_ptr ogr_index_featureset<filterT>::next() [with filterT = mapnik::filter_at_point; mapnik::feature_ptr = std::shared_ptr<mapnik::feature_impl>]’:
plugins/input/ogr/ogr_index_featureset.cpp:194:16:   required from here
plugins/input/ogr/ogr_index_featureset.cpp:140:13: warning: enumeration value ‘OFTInteger64’ not handled in switch [-Wswitch]
plugins/input/ogr/ogr_index_featureset.cpp:140:13: warning: enumeration value ‘OFTInteger64List’ not handled in switch [-Wswitch]
plugins/input/ogr/ogr_index_featureset.cpp:140:13: warning: enumeration value ‘OFTMaxType’ not handled in switch [-Wswitch]
```

**Comment by springmeyer** [comment]
Great, thanks for flagging.

**Comment by springmeyer** [comment]
just replicated a test failure that is likely due to needing handling of these new types: #2736

**Comment by springmeyer** [comment]
fixed in 093d857 (and confirmed this fixes the failing test mentioned at #2736).
